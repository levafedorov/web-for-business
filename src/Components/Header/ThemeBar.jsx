import React, {Component} from 'react';
import ThemeContext from '../PageContexts/PageThemes';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Button from '../Buttons/Button';
import PropTypes from 'prop-types';


export default class ThemeBar extends Component{
   
 static contextType = ThemeContext;
 
 btnHanndler = (ev) => {
      const theme = ev.currentTarget.dataset.theme;
      this.context.change(theme);
 }


    render(){
       const {btnHanndler} = this;
        return <div className='theme-bar'>
            <Button theme='default' fn={btnHanndler} class={'rounded small '}><FontAwesomeIcon icon="circle" color="grey" size="xs"/></Button>
            <Button theme='dark'   fn={btnHanndler} class={'rounded small '}><FontAwesomeIcon icon="moon" color="grey"/></Button>
            <Button theme='light' fn={btnHanndler} class={'rounded small '}><FontAwesomeIcon icon="sun" color="grey"/></Button>
        </div>
    }
}


