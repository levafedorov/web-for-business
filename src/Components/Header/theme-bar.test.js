import React from 'react';
import {shallow} from 'enzyme';
import ThemeBar from './ThemeBar';


test('Should ThemeBar render', () => {
    const component = shallow(<ThemeBar />);
    expect(component).toMatchSnapshot();
});




