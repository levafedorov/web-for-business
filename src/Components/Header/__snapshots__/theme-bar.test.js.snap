// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ThemeBar render 1`] = `
<div
  className="theme-bar"
>
  <Button
    class="rounded small "
    fn={[Function]}
    theme="default"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="grey"
      fixedWidth={false}
      flip={null}
      icon="circle"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="xs"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Button>
  <Button
    class="rounded small "
    fn={[Function]}
    theme="dark"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="grey"
      fixedWidth={false}
      flip={null}
      icon="moon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Button>
  <Button
    class="rounded small "
    fn={[Function]}
    theme="light"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="grey"
      fixedWidth={false}
      flip={null}
      icon="sun"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Button>
</div>
`;
